src/preprocessor.py

```diff
@@ -0,0 +1,80 @@
+import numpy as np
+import cv2 as cv
+import os
+
+
+"""Requires: opencv-python, numpy"""
+class Preprocessor:
+    def __init__(
+        self,
+        contrast: float, invert_colors: bool,
+        denoise_intensity: int,
+        sharpen_intensity: float,
+        input_path: str, output_path: str = None
+    ):
+        # Image processing settings
+        self.contrast = contrast
+        self.invert_colors = invert_colors
+        self.denoise_intensity = denoise_intensity
+        self.sharpen_intensity = sharpen_intensity
+
+        # Path variables
+        self.input_path  = input_path
+        self.output_path = output_path if output_path else input_path
+    
+    def process(self, image_file):
+        # Load initial image
+        img = cv.imread(image_file, cv.IMREAD_GRAYSCALE)
+        img = np.array(img)
+
+        # Brightness & Contrast adjustment
+        img = cv.convertScaleAbs(img, 0, self.contrast)
+
+        # Denoiser
+        img = cv.fastNlMeansDenoising(img, None, self.denoise_intensity, 50, 20)
+
+        # Sharpen filter
+        kernel = np.array([
+            [1,  -1, -1],
+            [-1, self.sharpen_intensity, -1],
+            [-1, -1, -1]
+        ])
+        img = cv.filter2D(img, 0, kernel)
+
+        """ 
+            Warning: Do not use, sh*t
+            Morphological operations to remove noise and fill gaps
+        """
+        #kernel = np.ones((2, 2), np.uint8)
```

Comment:
Do we need the commented lines or we can use them depending on some input parameter?

Reply:
You can play with those, makes letter way thicker, might help in some rare occasions